### ISSUE TITLE
Support for whole country as location

### ISSUE THREAD

**Issue by marcselman** [ecg-icas/icas#9]
In the Admarkt back-end, when editing an ad, I can choose between entering a postal code as location or 'Heel Nederland' (all of The Netherlands).
See: http://screencast.com/t/DJHU7UMO0cK
But in the API postcode is mandatory.

What value can I pass to have the ad show 'Heel Nederland'?

**Comment by twessling-icas** [comment]
Hi Marc,

The "Heel Nederland" functionality can be achieved by still setting a postcode, but selecting shipping type SHIP. This way, you will still be shown as "Bezorgt in heel Nederland" on marktplaats. Feel free to test this yourself on the test environment.

**Comment by marcselman** [comment]
Klopt, in de listing wordt dan wel "Bezorgt in xxx" getoond maar ik bedoel op de detail pagina.
Daar wordt bij de aanbieder gegevens altijd het adres adhv de postcode getoond. Maar het is ook mogelijk daar "Heel Nederland" te tonen maar volgens mij niet via de API.
Zie: http://screencast.com/t/OOKcqbW7uPD

**Comment by twessling-icas** [comment]
Hey Marc,

klopt dat dit niet via de API te zetten is. De weergave op de detail pagina hangt ook af van de ondersteuning van 'verzenden' in de categorie - kunst, auto's en huizen bijvoorbeeld ondersteunen dat niet. Welke categorie plaats je de advertentie? Dan probeer ik dat te reproduceren en kijken of daar iets aan gedaan kan worden.

**Comment by marcselman** [comment]
Hi,
In dit geval ging het inderdaad om een klant met woningen.
Juist omdat hij woningen uit heel nederland aanbied vind hij het niet mooi dat bij zijn profiel een plaatsnaam staat.

**Comment by twessling-icas** [comment]
Aha, I see. Zijn profiel pagina (https://www.marktplaats.nl/mymp/profiel/index.html) heeft ook geen postcode?

**Comment by marcselman** [comment]
Dit is een voorbeeld advertentie van de desbetreffende klant:
http://www.marktplaats.nl/a/zakelijke-goederen/bedrijfs-onroerend-goed/a1019268789-te-huur-bedrijfsruimte-bergen-op-zoom-poortweg-4.html?c=9b26ed2a557deff636f4f8b9c5b7a618

Zoals je ziet staat er nu "Barendrecht, Zuid-Holland" bij, maar de klant zou daar graag "Heel Nederland" zien.

Is dit iets wat aan de API kan of zal toegevoegd worden?

**Comment by twessling-icas** [comment]
Hi Marc,

deze advertentie heeft een postcode, waarschijnlijk omdat hij via de api gemaakt is?
Hoe dan ook, wij zijn zelf ook niet helemaal tevreden met de manier waarop locaties behandeld worden, we willen dit graag aanpakken. We zullen deze optie zeker meenemen daarin, het is namelijk voor Nederland zeer toepasselijk. Helaas kan ik je geen inschatting geven wanneer we dit gaan doen, maar ik verwacht wel op redelijke termijn.

**Comment by marcselman** [comment]
Klopt, wij plaatsen alle advertenties van onze klanten altijd via de API.
Ik zal de ticket open laten totdat de optie geimplementeerd is.

Bedankt.

**Comment by barbarap86** [comment]
Goedemiddag,

Deze vraag staat nog open voor onze klant Huurbieding. Is er al enig idee wanneer deze optie wordt geïmplementeerd voor API gebruikers?

**Comment by magiconair** [comment]
Not yet but I'll have another look at how difficult it is to implement it.

**Comment by barbarap86** [comment]
Thanks for looking into this!
